### PR TITLE
asset: Add extrinsic documentation 

### LINKED ISSUE #443: [C4GT] Add extrinsic documentation
## Description
Improve documentation for extrinsics with brief writeup on what it does, parameters it takes, what it returns & errors if emitted.

## Goals

- [ ] create
- [ ] vc_create
- [ ] issue
- [ ] vc_issue
- [ ] transfer
- [ ] vc_transfer
- [ ] status_change
- [ ] vc_status_change

## Expected Outcome
- [ ] Should have the method documentation in the below format
```
/// Adds a delegate with the ability to assert new entries to a space.
///
/// The `ASSERT` permission allows the delegate to sign and add new
/// entries within the space. This function is called to grant a
/// delegate this specific permission. It checks that the caller has the
/// necessary authorization (admin rights) to add a delegate to the
/// space. If the caller is authorized, the delegate is added with the
/// `ASSERT` permission using the `space_delegate_addition`
/// internal function.
///
/// # Parameters
/// - `origin`: The origin of the call, which must be signed by an admin of the space.
/// - `space_id`: The identifier of the space to which the delegate is being added.
/// - `delegate`: The identifier of the delegate being added to the space.
/// - `authorization`: The authorization ID used to validate the addition.
///
/// # Returns
/// Returns `Ok(())` if the delegate was successfully added with
/// `ASSERT` permission, or an `Err` with an appropriate error if the
/// operation fails.
///
/// # Errors
/// - `UnauthorizedOperation`: If the caller is not an admin of the space.
/// - Propagates errors from `space_delegate_addition` if it fails.
#[pallet::call_index(0)]
#[pallet::weight(<T as pallet::Config>::WeightInfo::add_delegate())]
pub fn add_delegate(
```

## Story Points
2

## Time Constraint
3 Days

## Mockups / Wireframes
NA

## Implementation Details
Go to file `pallets/asset/src/lib.rs` and write the documentation.

---- 

## Product Name
CORD

## Organization Name
Dhiway

## Domain
Blockchain

## Tech Skills Needed
Rust

## Mentor(s)
@vatsa287 @amarts @NiranjanAP 

## Complexity
[Low]

## Category
[Documentation]

## Sub Category
[Beginner friendly]

## NOTE:
Contributors can submit to a part or for all of the required methods.
Assignees will be re-assigned if no PRs are opened after 3 days.

### PR DESCRIPTION
**Pull request by vidyaa01**
## Comprehensive Documentation Update

This PR introduces comprehensive documentation for the following extrinsic functions in our pallet:

- create
- issue
- transfer
- status_change
- vc_create
- vc_issue
- vc_transfer
- vc_status_change

Each function now includes detailed Rustdoc comments covering:

- Purpose: What the function does.
- Parameters: Descriptions of input parameters.
- Return Values: Information on what the function returns.
- Potential Errors: Possible errors that can be encountered.
- Events Emitted: Events triggered by the function.

Fixes: #443

### CODE REVIEW

File: pallets/asset/src/lib.rs [hunk 1]
```diff
@@ -300,6 +300,34 @@ pub mod pallet {
 
 			Ok(())
 		}
+/// Creates a new asset entry within a specified space.
```
Comment (vatsa287) [inline pallets/asset/src/lib.rs]: Please move the method documentation to top of the actual method which the doc is targeted for & match its indentation.

Comment (vidyaa01) [reply pallets/asset/src/lib.rs]: Please review it again, as I have updated the documentation accordingly.

**Review by vatsa287 — DISMISSED** [review]
LGTM

**Comment by vatsa287** [comment]
@vidyaa01 Run the cargo format for the ci test to pass.

```
cargo +nightly fmt --all
```

Add the changes again, amend the commit and push the changes.

**Comment by vidyaa01** [comment]
Hey @vatsa287 , 

 I have made the changes you requested and pushed the changes to the PR. It shows me dismissed stale review which wasn't my intention, can you guide me on further steps?

**Review by vatsa287 — APPROVED** [review]
LGTM

**Comment by vatsa287** [comment]
@amarts PTAL

**Review by amarts — APPROVED** [review]

**Comment by VedantKhairnar** [comment]
Hello @vidyaa01   ,
Thanks for your contribution.
As part of Augtoberfest, we want to reward you for your contributions, so pls fill out [this form](https://forms.gle/VBDUPWMYcZEbmVrQ9) by 19th Sept.
Thanks.

**Pull request merged**